### ISSUE TITLE
[BUG] [Service] [Component-based API] Incorrect response status when translation no exist and checkTranslationStatus=true

### ISSUE THREAD

**Issue by nannany00** [vmware/singleton#1425]
**Describe the bug**
In Singleton Service, here is a feature (checkTranslationStatus=true) that to return translation status (1-ready or 0-not-ready), it works well for Key-based APi; but for component-based API, it still returns translation ready even if the translation not exist but the source value is same between messages_en.json and messages_latest.json.

**To Reproduce**
1. Go to Service API: `GET /i18n/api/v2/translation/products/{productName}/versions/{version}/locales/{locale}/components/{component} Get a component's translation`
2. set `checkTranslationStatus=true`
3. Check status for one key
`Request message data`
```
latest:
{
"a" : "a",
"b" : "b",
"c" : "c",
"d": "d"
}
en:
{
"a" : "a",
"b" : "b",
"c" : "c"
}
de:
{
"a" : "a",
"b" : "b"
}
```
Actual Response:
```
{
    "response":{
        "code":205,
        "message":"Collected resource's translations is not ready",
        "serverTime":""
    },
    "signature":"",
    "data":{
        "productName":"Testing",
        "version":"1.0.0",
        "dataOrigin":"bundle",
        "pseudo":false,
        "machineTranslation":false,
        "component":"default",
        "messages":{
            "a":"a",
            "b":"b"
        },
        "locale":"de",
        "status":"{\"a\":\"1\",\"b\":\"1\",\"c\":\"1\",\"d\":\"0\"}",
        "id":0
    }
}
```
**Expected behavior**
The translation status that not existing in request locale should also be 0-not-ready.

**Comment by linr211** [comment]
How this issue impact the product? and why it's high priority?

**Comment by InaSong** [comment]
Assign to Tiger for first round of triage.

**Comment by lyiyu66** [comment]
The issue is fixed with commit 5f9c93ce435a14118373688d45c3f27d5b2d3f7a
Please see the screenshots: 
![image](https://user-images.githubusercontent.com/51813046/180691409-d4cc7a4e-616f-4ae1-b5c4-1568fb87b343.png)
